File: frontend/src/lib/features/auth/member.ts
import z from "zod";
import { teamSchema } from "./team";
import { throwAPIError } from "@/lib/api/error";

export const memberSchema = z.object({
	id: z.int().positive(),
	email: z.email().nonempty(),
	firstName: z.string().nonempty(),
	lastName: z.string().nonempty(),
	avatarUrl: z.url().nonempty(),
	specialRole: z.string().optional(),
	approved: z.boolean(),
	teams: teamSchema.array(),
})

export type Member = z.infer<typeof memberSchema>

File: backend/member.go
package konnekt

type Member struct {
	ID          int64  `json:"id"`
	Email       string `json:"email"`
	FirstName   string `json:"firstName"`
	LastName    string `json:"lastName"`
	AvatarURL   string `json:"avatarUrl"`
	SpecialRole string `json:"specialRole,omitempty"`
	Teams       []Team `json:"teams"`
	Approved    bool   `json:"approved"`
}

type CreateMember struct {
	Email     string `json:"email"`
	FirstName string `json:"firstName"`
	LastName  string `json:"lastName"`
	AvatarURL string `json:"avatarUrl"`
	Password  string `json:"password"`
}
